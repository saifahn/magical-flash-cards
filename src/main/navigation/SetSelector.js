import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media, text } from '../../utils/theme';

const Selector = styled.select`
  width: 100%;
  padding: 0.25rem 0 0.25rem 0.5rem;
  font-size: ${text.lg};
  font-family: 'Lora', serif;
  border-radius: 0px;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: rgba(0, 0, 0, 0.1);
  outline: none;
  margin: auto;

  ${media.sm`
    font-size: ${text.lg};
  `}

  ${media.md`
    font-size: ${text.xl};
  `}

  ${media.lg`
    max-width: 560px;
  `}
  
  option {
    font-size: ${text.md};
  }
`;

class SetSelector extends Component {
  handleSetChange = (e) => {
    const { handleSetChange } = this.props;
    handleSetChange(e.target.value);
  }

  render() {
    const { sets } = this.props;
    return (
      <Selector className="some-class" onChange={this.handleSetChange}>
        {sets.map(set => (
          <option key={set.code} value={set.code}>
            {set.name}
          </option>
        ))}
      </Selector>
    );
  }
}

SetSelector.propTypes = {
  handleSetChange: PropTypes.func.isRequired,
  sets: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default SetSelector;
