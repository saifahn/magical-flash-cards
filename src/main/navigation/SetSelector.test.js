/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import SetSelector from './SetSelector';

test('it renders correctly', () => {
  const wrapper = shallow(<SetSelector />);
  expect(wrapper).toMatchSnapshot();
});
