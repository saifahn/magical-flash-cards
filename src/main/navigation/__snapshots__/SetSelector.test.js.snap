// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<select
  onChange={[Function]}
>
  <option
    value="M19"
  >
    Core Set 2019
  </option>
  <option
    value="DOM"
  >
    Dominaria
  </option>
  <option
    value="RIX"
  >
    Rivals of Ixalan
  </option>
  <option
    value="XLN"
  >
    Ixalan
  </option>
</select>
`;
